src/main/java/com/wolox/training/services/OpenLibraryService.java

```diff
@@ -0,0 +1,50 @@
+package com.wolox.training.services;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.wolox.training.dtos.BookInfoDTO;
+import com.wolox.training.exceptions.MyJsonProcessingException;
+import java.util.HashMap;
+import java.util.Optional;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+public class OpenLibraryService {
+
+    private static final String BASE_URL = "https://openlibrary.org/api/books";
+
+    public Optional<BookInfoDTO> bookInfo(String isbn) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.setVisibility(PropertyAccessor.ALL, Visibility.NONE);
+        objectMapper.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
+        objectMapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+        RestTemplate restTemplate = new RestTemplate();
+
+        String url = new StringBuilder(BASE_URL).append("?bibkeys=ISBN:").append(isbn)
```

Comment:
Maybe you could check the String.format() method for this url. This way you don't need to append so many times and just set the variable into the url string using %s.